### PR TITLE
Banner management

### PR DESCRIPTION
**Pull request by insacuri**
# Motivation and Context

# What has changed

# How to test?

# Links

### CODE REVIEW

**Review by warrenbailey — CHANGES_REQUESTED** [review]

File: response_operations_ui/views/admin.py [hunk 1]
```diff
@@ -1,111 +1,174 @@
...
 
 @admin_bp.route('/banner', methods=['POST'])
 @login_required
-def update_banner():
+def post_banner():
+    if request.form.get('delete'):
+        # Handle remove scenario
+        logger.info("Removing active status from banner")
+        admin_controller.remove_banner()
+        flash('The alert has been removed')
```
[the remark this answers is not in the document]
Comment (insacuri) [reply response_operations_ui/views/admin.py:50]: This one is in the design.  I'll mention this along with the use of the word alert or banner

File: response_operations_ui/views/admin.py [hunk 2]
```diff
@@ -1,111 +1,174 @@
...
+                           breadcrumbs=breadcrumbs)
 
 
 @admin_bp.route('/banner', methods=['POST'])
 @login_required
-def update_banner():
+def post_banner():
+    if request.form.get('delete'):
+        # Handle remove scenario
+        logger.info("Removing active status from banner")
```
[the remark this answers is not in the document]
Comment (insacuri) [reply response_operations_ui/views/admin.py]: Yup, done

File: response_operations_ui/views/admin.py [hunk 3]
```diff
@@ -1,111 +1,174 @@
...
+    form = BannerPublishForm(form=request.form)
+    banner_text = form.banner_text.data
+    logger.info("Setting an active banner", user=current_username(), banner_text=banner_text)
+    if banner_text:
+        admin_controller.set_banner(banner_text)
+        return redirect(url_for("surveys_bp.view_surveys", message_key='alert_published'))
     else:
-        return redirect(url_for("admin_bp.banner_admin"))
+        logger.error("Banner text is somehow missing from the form")
+        flash("Banner text somehow not found on publish.  Please try again.")
```
[the remark this answers is not in the document]
Comment (insacuri) [reply response_operations_ui/views/admin.py]: Done

File: response_operations_ui/views/admin.py [hunk 4]
```diff
@@ -1,111 +1,174 @@
...
-                               form=form,
-                               current_banner=current_banner,
-                               breadcrumbs=breadcrumbs,
-                               time_banner_set=time_banner_set)
+
+    form = BannerPublishForm(form=request.form)
+    try:
+        banner_text = session.pop('banner-text')
+    except KeyError:
+        flash("Something went wrong setting the banner")
```
[the remark this answers is not in the document]
Comment (insacuri) [reply response_operations_ui/views/admin.py]: That's fair.  I went back and fourth on the wording but the simple one is best I think.

[the remark this answers is not in the document]
Comment (insacuri) [reply response_operations_ui/views/admin.py]: Done

File: response_operations_ui/views/admin.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (insacuri) [reply response_operations_ui/views/admin.py]: Done

File: response_operations_ui/views/admin.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (insacuri) [reply response_operations_ui/views/admin.py:50]: We're going with alert.  I've changed all uses of the word banner in the text to alert

**Review by warrenbailey — APPROVED** [review]

**Pull request merged**